apps/studio/src/pages/sites/[siteId]/settings/notification.tsx

```diff
@@ -0,0 +1,137 @@
+import { NotificationSchema } from "@opengovsg/isomer-components"
+import { ResourceType } from "~prisma/generated/generatedEnums"
+import { isEqual } from "lodash"
+import { BiWrench } from "react-icons/bi"
+
+import { PermissionsBoundary } from "~/components/AuthWrappers"
+import { ISOMER_SUPPORT_EMAIL } from "~/constants/misc"
+import { BRIEF_TOAST_SETTINGS } from "~/constants/toast"
+import { EditSettingsPreview } from "~/features/editing-experience/components/EditSettingsPreview"
+import { ErrorProvider } from "~/features/editing-experience/components/form-builder/ErrorProvider"
+import FormBuilder from "~/features/editing-experience/components/form-builder/FormBuilder"
+import { UnsavedSettingModal } from "~/features/editing-experience/components/UnsavedSettingModal"
+import { siteSchema } from "~/features/editing-experience/schema"
+import { SettingsEditingLayout } from "~/features/settings/SettingsEditingLayout"
+import { SettingsHeader } from "~/features/settings/SettingsHeader"
+import { useNavigationEffect } from "~/hooks/useNavigationEffect"
+import { useNewSettingsPage } from "~/hooks/useNewSettingsPage"
+import { useQueryParse } from "~/hooks/useQueryParse"
+import { type NextPageWithLayout } from "~/lib/types"
+import { Notification } from "~/schemas/site"
+import { SiteSettingsLayout } from "~/templates/layouts/SiteSettingsLayout"
+import { ajv } from "~/utils/ajv"
+import { trpc } from "~/utils/trpc"
+
+const NotificationSettingsPage: NextPageWithLayout = () => {
+  const isEnabled = useNewSettingsPage()
+  const router = useRouter()
+  const { siteId: rawSiteId } = useQueryParse(siteSchema)
+  const siteId = Number(rawSiteId)
+  const trpcUtils = trpc.useUtils()
+  const toast = useToast()
+  const [{ name }] = trpc.site.getSiteName.useSuspenseQuery({
+    siteId: Number(siteId),
+  })
+  const validateFn = ajv.compile<Notification>(NotificationSchema)
+
+  const notificationMutation = trpc.site.setNotification.useMutation({
+    onSuccess: async () => {
+      await trpcUtils.site.getNotification.invalidate({ siteId })
+      toast({
+        title: "Saved site notification!",
+        description: "Check your site in 5-10 minutes to view it live.",
+        status: "success",
+        ...BRIEF_TOAST_SETTINGS,
+      })
+    },
+    onError: () => {
+      toast({
+        title: "Error saving site notification!",
+        description: `If this persists, please report this issue at ${ISOMER_SUPPORT_EMAIL}`,
+        status: "error",
+        ...BRIEF_TOAST_SETTINGS,
+      })
+    },
+  })
+
+  const [{ notification: previousNotification }] =
+    trpc.site.getNotification.useSuspenseQuery({
+      siteId,
+    })
+  const [state, setState] = useState<Notification>(
+    previousNotification ?? {
+      enabled: false,
+      title: "",
+      content: { type: "prose", content: [] },
+    },
+  )
+
+  useEffect(() => {
+    if (!isEnabled) {
+      router.push(`/sites/${siteId}/settings`)
```

Comment:
issue: seems like there is a redirect loop with the original settings page

Reply:
leaving this as is first, confiring w rachel that we're ok with rolling out all at once. if that's the case, i'll remove all the flags and this should be fixed